hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/WriteStatus.java

```diff
@@ -181,6 +189,21 @@ private void updateStatsForFailure() {
     totalErrorRecords++;
   }
 
+  public WriteStatus removeMetadataIndexStatsAndErrorRecordsTracking() {
+    removeMetadataStats();
+    dropGranularErrorRecordsTracking();
+    return this;
+  }
+
+  public WriteStatus removeMetadataStats() {
+    this.writtenRecordDelegates.clear();
+    return this;
+  }
+
+  public void dropGranularErrorRecordsTracking() {
+    failedRecords.clear();
+  }
```

Comment:
Can these be called independently besides `removeMetadataIndexStatsAndErrorRecordsTracking()`?

Reply:
Yes these are used currently in SparkRDDWriteClient